src/System.Runtime.Loader/tests/DefaultContext/DefaultLoadContextTest.cs

```diff
@@ -58,24 +58,43 @@ public static void LoadInDefaultContext()
         {
             Init();
 
-           // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
-           var assemblyName = "System.Runtime.Loader.Noop.Assembly";
+            // This will attempt to load an assembly, by path, in the Default Load context via the Resolving event
+            var assemblyNameStr = "System.Runtime.Loader.Noop.Assembly";
+            var assemblyName = new AssemblyName(assemblyNameStr);
 
+            // By default, the assembly should not be found in DefaultContext at all
+            Assert.Throws(typeof(FileNotFoundException), () => Assembly.Load(assemblyName));
+
+            // Now, wireup the Resolving event to locate the assembly
             AssemblyLoadContext.Default.Resolving += ResolveAssembly;
-            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(new AssemblyName(assemblyName));
+            
+            // This will invoke the resolution via VM requiring to bind using the TPA binder
+            var assemblyExpectedFromLoad = Assembly.Load(assemblyName);
+
+            // We should have successfully loaded the assembly in default context.
+            Assert.NotNull(assemblyExpectedFromLoad);
+
+            // And make sure the simple name matches
+            Assert.Equal(assemblyNameStr, assemblyExpectedFromLoad.GetName().Name);
+
+            // This will resolve the assembly via event invocation.
+            var assemblyExpected = AssemblyLoadContext.Default.LoadFromAssemblyName(assemblyName);
             
             // We should have successfully loaded the assembly in default context.
             Assert.NotNull(assemblyExpected);
 
+            // What we got via Assembly.Load and LoadFromAssemblyName should be the same
+            Assert.Equal(assemblyExpected, assemblyExpectedFromLoad);
+
             // And make sure the simple name matches
-            Assert.Equal(assemblyExpected.GetName().Name, assemblyName);
+            Assert.Equal(assemblyExpected.GetName().Name, assemblyNameStr);
 
             // Unwire the Resolving event.
             AssemblyLoadContext.Default.Resolving -= ResolveAssembly;
```

Comment:
From an end user's perspective, it doesn't matter where in the runtime the ordering comes from: there is an order, and code can inadvertently take a dependency on it (unless you're saying that every invocation returns the results in random order?).  This wouldn't be as big an issue if this was a normal void-returning event handler, but it's not, and invocation of registered handlers stops the moment one registered handler returns non-null, making the ordering very observable and impacting behavior.

Again, if we did break this in the future, I'm not saying we would choose to revert that, but at least knowing that we broke it helps to make an informed decision about impact.  We could trivially make a change that would change the order (e.g. iterating from end to beginning instead of beginning to end), change exception handled (e.g. wrap each invocation in a try/catch), stop short-circuiting (e.g. invoking each delegate and then only returning a result after all were invoked), etc., all of which would have a visible impact on app behavior, but our tests wouldn't catch it.

If it were my code, I'd add tests for this.  But I'll drop it at this point.